packages/backend/src/projectAdapters/dbtBaseProjectAdapter.ts

```diff
@@ -124,10 +126,19 @@ export class DbtBaseProjectAdapter implements ProjectAdapter {
 
         // Be lazy and try to attach types to the remaining models without refreshing the catalog
         try {
+            const warehouseCatalog =
+                await this.projectModel.getWarehouseFromCache(projectUuid);
```

Comment:
@owlas @rephus fyi: This breaks our initial architecture rules where models where only used in services and adapters didn't depend on our Database. We had a plan for the adapters to be stand alone entities and later be extracted to their own 'adapters' package.

Reply:
what do you suggest to fix this ? should I include a method in `projctService` and call that instead ?

Reply:
@ZeRego good point, is there a better alternative you see? I found the boundary between the adapter and service to quite fuzzy

For me it was either:
1. Pull logic outside the adapter into the service (leaving not much left in the adapter?)
2. Add this dependency for caching

Reply:
We accept the architecture changed and they are no longer standalone entities **or** we would need to get the cached warehouse catalog in the project service and a callback to save the new cache. 

```
const cachedWarehouseCatalog =
                await this.projectModel.getWarehouseFromCache(projectUuid);

const adapter = await projectAdapterFromConfig(
            project.dbtConnection,
            project.warehouseConnection,
            { // new options object 
                warehouseCatalog: cachedWarehouseCatalog,
                onWarehouseCatalogChange: async (warehouseCatalog) => {
                     await this.projectModel.saveWarehouseToCache(
                         projectUuid,
                         warehouseCatalog,
                    );
                } 
            }
        );
```

Reply:
refactor here : https://github.com/lightdash/lightdash/pull/1881/commits/077d004081718ebb767b403df1bbef9a4fba3669 @ZeRego is this what you had in mind ?